src/main/java/dev/codesupport/web/common/service/controller/AuthenticationController.java

```diff
@@ -27,36 +29,67 @@
 
     private AuthenticationManager authenticationManager;
     private AuthenticationUserDetailsService userDetailsService;
+    private JwtUtility jwtUtility;
 
     @Autowired
     public AuthenticationController(
             AuthenticationManager authenticationManager,
-            AuthenticationUserDetailsService userDetailsService
+            AuthenticationUserDetailsService userDetailsService,
+            JwtUtility jwtUtility
     ) {
         this.authenticationManager = authenticationManager;
         this.userDetailsService = userDetailsService;
+        this.jwtUtility = jwtUtility;
     }
 
+    /**
+     * Handles authentication post requests
+     * <p>Accepts posts with a {@link AuthenticationRequest} body, authenticating the credentials and
+     * returning a JWT if valid.
+     * If authentication fails, an exception is thrown, bubbling back up and resulting in a call
+     * to the {@link ErrorHandlerController}</p>
+     *
+     * @param authenticationRequest The {@link AuthenticationRequest} body
+     * @return The encoded JWT string if authentication passes.
+     */
     @PostMapping(value = "/authenticate")
     public ResponseEntity<RestResponse<Serializable>> createAuthenticationToken(@RequestBody AuthenticationRequest authenticationRequest) {
         authenticate(authenticationRequest.getUsername(), authenticationRequest.getPassword());
         final UserDetails userDetails = userDetailsService
                 .loadUserByUsername(authenticationRequest.getUsername());
```

Comment:
Cache `AuthenticationRequest.getUsername()` as it is called multiple times.

Reply:
Good point!

Reply:
made this change